server/api/lib/mavat/index.js

```diff
@@ -39,22 +39,99 @@ const init = () =>
 		})();
 	});
 
-const downloadFile = (url, file, entityDocId, entityDocNumber) => {
+const downloadChallengedFile = (url, file, options) => {
 	return new Promise((resolve) => {
-		http.get(url, (response) => {
-			response.pipe(file);
-			file.on('finish', async function () {
-				await file.close();
-				resolve(true);
-			});
+		options = options || {};
+
+		http.get(url, options, (response) => {
+			if (response.statusCode !== 200) {
+				Log.info(`downloadChallengedFile failed with status ${response.statusCode} for url ${url}`);
+				resolve(false);
```

Comment:
Why not reject instead of resolving true?

Reply:
i was trying to preserve the interface and just replace the single download function